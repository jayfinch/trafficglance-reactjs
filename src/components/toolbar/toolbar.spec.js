import React from 'react'
import Toolbar from './toolbar'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

describe('<Toolbar />', () => {
  describe('render', () => {
    it('Should handle clicking refresh-all button', () => {
      const mock = jest.fn()
      const component = <Toolbar
        onClickRefreshAll={mock}
        isRefreshAvailable
      />

      const wrapper = shallow(component)
      wrapper.find('.toolbar__refresh-all').simulate('click', { preventDefault () {} })

      expect(mock).toHaveBeenCalled()
    })

    it('Should render with refresh button', () => {
      const mock = jest.fn()
      const tree = renderer.create(
        <Toolbar
          onClickRefreshAll={mock}
          isRefreshAvailable
        />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    it('Should render without refresh button', () => {
      const mock = jest.fn()
      const tree = renderer.create(
        <Toolbar
          onClickRefreshAll={mock}
        />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
