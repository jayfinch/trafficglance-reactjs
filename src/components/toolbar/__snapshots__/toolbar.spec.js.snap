exports[`<Toolbar /> render Should render with refresh button 1`] = `
<div
  className="toolbar">
  <div
    className="toolbar__site-name">
    TrafficGlance
  </div>
  <button
    className="toolbar__refresh-all"
    onClick={[Function]}>
    Refresh All
  </button>
</div>
`;

exports[`<Toolbar /> render Should render without refresh button 1`] = `
<div
  className="toolbar">
  <div
    className="toolbar__site-name">
    TrafficGlance
  </div>
</div>
`;
